velox/exec/TableWriter.h

```diff
@@ -16,11 +16,52 @@
 
 #pragma once
 
+#include "velox/connectors/WriteProtocol.h"
 #include "velox/core/PlanNode.h"
 #include "velox/exec/Operator.h"
 
 namespace facebook::velox::exec {
 
+class TableWriterWriteInfo : public connector::WriteInfo {
```

Comment:
Is this used only in the .cpp file? Should we move it there?

Reply:
> @gggrace14 Have some questions. I'm seeing that HiveTaskCommitWriteProtocol::commit uses default implementation. Who is using the information in the TableWriteWriteInfo?

Hi @mbasmanova , I have PrestoNoCommitWriteProtocol and PrestoTaskCommitWriteProtocol in the native repo, and they use the TableWriterWriteInfo. https://github.com/prestodb/presto/pull/18377/files#diff-d18ada7dc9e1ebc5521dd0f7e48e7df856c3474513a7320f3ee0680042ba3065R123-R126

TableWriterWriteInfo is the write info that TableWriter operator needs to pass to write protocols. So it's TableWriter's implementation. It's possible that a different operator could also have write info to pass to write protocols. Maybe I call it CommitInfo?